doc/adr/0010-leaf-cluster-k8s-auth.md

```diff
@@ -0,0 +1,63 @@
+# 10. Leaf Cluster Kubernetes API Authentication
+
+Date: 2021-11-16
+
+## Status
+
+Accepted
+
+## Context
+
+When a user "onboards" a new Leaf Cluster, we need ensure that the Weave GitOps API deployment running on the Management Cluster can query data about the objects on the Leaf Cluster. At present, we would have to rely on the user's local `kubeconfig` to provide all necessary authentication credentials to query multiple clusters at once.
+
+This approach won't work for most use cases, as we cannot expect every user of the WeGO UI (and subsequently API) to have access or authorization to view objects on all clusters.
+
+## Design
+
+To allow a user to query data about a Leaf Cluster, we must do (roughly) these steps during the "onboarding" CLI command:
+
+1. Create a `ServiceAccount` on the Leaf Cluster with cluster admin priveleges
+2. Add the `kubeconfig` for that `ServiceAccount` as a `Secret` on the Management Cluster.
+3. Add a `Cluster` Custom Resource to the Management Cluster to persist the `Cluster` record for future retreival
+
+**Assumptions**:
+
+- The user doing `gitops install` is a cluster admin and can create a cluster admin `ServiceAccount`
+
+#### Leaf Cluster Sequence Diagrams
+
+Here are the steps that need to happen during the Leaf Cluster onboarding.
+
+![Leaf Cluster Installation Diagram](leaf-cluster-install.svg)
+
+Then, when a user makes a request to the UI(+API) running on the Management Cluster, it looks like this (written as psuedo-code):
+
+![Leaf Cluster Request Diagram](leaf-cluster-request.svg)
+
+#### RBAC
+
+Access to the Leaf Clusters and Applications will be controlled by setting permissions on the individual `Cluster` and `Application` resources themselves. Users will use the existing Kubernetes RBAC machinery to manage these permissions.
+
+#### Cluster Custom Resource Definition
+
+A `Cluster` resource might look something like this (note that this is the resource itself, not the CRD):
+
+```yaml
+apiVersion: wego.weave.works/v1alpha1
+kind: Cluster
+metadata:
+  name: my-cluster
+  namespace: wego-system
+spec:
+  kubeconfig_secret_name: my-cluster-kubeconfig
+  kubeconfig_secret_namespace: wego-system
+  config_repo_url: ssh://git@gitlab.com/jpellizzari1/my-repo.git
+```
```

Comment:
I like the CR model.

I suspect the keys should use the "camelcase" convention, rather than "snake", for example from K8s...

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx-demo
spec:
  containers:
  - name: nginx
    image: nginx:1.14.2
    ports:
    - containerPort: 80
```

Both `apiVersion` and `containerPort` are camel-case?